src/robotFactory.js

```diff
@@ -1,10 +1,84 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
```

Comment:
You simply assign a new object with all the fields. If you are given an empty object as a class argument, it will override your default object.

```suggestion
  constructor(name, weight, coords, chipVersion) {
```